.github/scripts/create-flaky-test-report.ts

```diff
@@ -0,0 +1,148 @@
+    workflow_id: env.WORKFLOW_ID,
+    branch: env.BRANCH,
+    event: 'push',
+    status: 'failure',
+    created: `${from.toISOString()}..${to.toISOString()}`,
+    per_page: 100,
+  });
+
+  const artifacts = (
+    await Promise.all(
+      runs.map(async (run) => {
+        const artifacts = await github.paginate(
+          github.rest.actions.listWorkflowRunArtifacts,
+          {
+            owner: env.OWNER,
+            repo: env.REPOSITORY,
+            run_id: run.id,
+            per_page: 100,
+          },
+        );
+
+        const chromeTestReport = artifacts.find(
+          (artifact) => artifact.name === 'test-e2e-chrome-report',
+        );
+        if (!chromeTestReport) return [];
+
+        const firefoxTestReport = artifacts.find(
+          (artifact) => artifact.name === 'test-e2e-firefox-report',
+        );
+        if (!firefoxTestReport) return [];
+
+        return [chromeTestReport, firefoxTestReport];
+      }),
+    )
+  ).flat();
+
+  const testRuns: TestRun[] = (
+    await Promise.all(
+      artifacts.map(async (artifact) => {
+        const response = await github.rest.actions.downloadArtifact({
+          owner: env.OWNER,
+          repo: env.REPOSITORY,
+          artifact_id: artifact.id,
+          archive_format: 'zip',
+        });
+        const zip = await JSZip.loadAsync(response.data as ArrayBuffer);
+        const file = zip.files['test-runs.json'];
+        if (!file) throw new Error(`'test-runs.json' file in zip not found!`);
+        const content = await file.async('string');
+        return JSON.parse(content);
+      }),
+    )
+  ).flat();
+
+  const failedTests = testRuns.flatMap((testRun) =>
+    testRun.testFiles.flatMap((testFile) =>
+      testFile.testSuites.flatMap((testSuite) =>
+        testSuite.testCases
+          .filter((testCase) => testCase.status === 'failed')
+          .map((testCase) => ({ ...testCase, path: testFile.path })),
+      ),
+    ),
+  );
+
+  const summarizedFailedTests = Object.values(
+    failedTests.reduce<
+      Record<string, (typeof failedTests)[number] & { count: number }>
+    >((summary, test) => {
+      if (summary[test.name]) {
+        summary[test.name].count += 1;
+      } else {
+        summary[test.name] = { ...test, count: 1 };
+      }
+      return summary;
+    }, {}),
+  ).sort((a, b) => b.count - a.count);
+
+  const options = { year: 'numeric', month: 'long', day: 'numeric' } as const;
+  const fromDateString = from.toLocaleDateString('en-US', options);
+  const toDateString = to.toLocaleDateString('en-US', options);
+
+  console.log(
+    `❌ Failed tests on the ${env.REPOSITORY} repository ${env.BRANCH} branch from ${fromDateString} to ${toDateString}`,
+  );
+  const lines = [
+    `\n\n:x: Failed tests on the <https://github.com/${env.OWNER}/${env.REPOSITORY}|${env.REPOSITORY}> repository <https://github.com/${env.OWNER}/${env.REPOSITORY}/tree/${env.BRANCH}|${env.BRANCH}> branch from ${fromDateString} to ${toDateString}`,
```

Comment:
same comment about using `new URL` as below, but maybe not as important since the env vars here can't be controlled by external contributors, since it is only run with the required secrets on a controlled cronjob.